### PR TITLE
Move most of the remaining Strong Name code to internal to the C++ file and remove duplicate validation

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Remove all function prototypes for functions only called in `strongnameinternal.cpp` from `strongnameinternal.h`.

Remove duplicate validation in `StrongNameTokenFromPublicKey`. These validation checks were already done in `StrongNameIsValidPublicKey`.

Remove extraneous `#include` of one of the key headers in `methodtablebuilder.cpp`.

### CODE REVIEW

File: src/coreclr/md/runtime/strongnameinternal.cpp [hunk 1]
```diff
@@ -6,20 +6,28 @@
...
 #include "strongnameinternal.h"
+#include "sha1.h"
+
+// We allow a special abbreviated form of the Microsoft public key (16 bytes
+// long: 0 for both alg ids, 4 for key length and 4 bytes of 0 for the key
+// itself). This allows us to build references to system libraries that are
+// platform neutral (so a 3rd party can build SPCL replacements). The
+// special zero PK is just shorthand for the local runtime's real system PK,
+// which is always used to perform the signature verification, so no security
+// hole is opened by this. Therefore we need to store a copy of the real PK (for
```
Comment (danmoseley) [inline src/coreclr/md/runtime/strongnameinternal.cpp:17]: Security hole? I thought these keys had no security purpose in Core.

Comment (jkoritzinsky) [reply src/coreclr/md/runtime/strongnameinternal.cpp:17]: I was just preserving the comment that was previously before the duplicate validation. I moved it up to before the copy of the validation I kept.

Comment (jkotas) [reply src/coreclr/md/runtime/strongnameinternal.cpp:17]: I think this comment can be deleted. It is not very useful.

File: src/coreclr/md/runtime/strongnameinternal.cpp [hunk 2]
```diff
@@ -6,20 +6,28 @@
...
+// this platform) here.
 #include "thekey.h"
```
Comment (jkotas) [inline src/coreclr/md/runtime/strongnameinternal.cpp:19]: You can simplify it even more and delete these header files. These keys are only ever used in the PublicKey -> PublickKeyToken conversions as a cache that skips over the more expensive SHA1 computation. These PublicKey and PublicKeyToken can be an internal implementation details of the PublicKey -> PublickKeyToken conversion method. They do not need to be distributed over multiple header files.

(While you are on it, you can also check that we have the cache pairs for all PublicKeys used by our shared frameworks. I won't be surprised if we have missed some. This cache should be good for startup perf.)

File: src/coreclr/inc/strongnameinternal.h [hunk 3]
```diff
@@ -15,16 +15,6 @@ typedef struct {
...
-DWORD StrongNameSizeOfPublicKey(const PublicKeyBlob &keyPublicKey);
-
-bool StrongNameIsValidPublicKey(_In_reads_(cbPublicKeyBlob) const BYTE *pbPublicKeyBlob, DWORD cbPublicKeyBlob);
-bool StrongNameIsValidPublicKey(const PublicKeyBlob &keyPublicKey);
-
-// Determine if a public key is the ECMA key
-bool StrongNameIsEcmaKey(_In_reads_(cbKey) const BYTE *pbKey, DWORD cbKey);
-bool StrongNameIsEcmaKey(const PublicKeyBlob &keyPublicKey);
-
 HRESULT StrongNameTokenFromPublicKey(BYTE* pbPublicKeyBlob,  // [in] public key blob
```
Comment (jkotas) [inline src/coreclr/inc/strongnameinternal.h:18]: StrongName token is always 8 bytes (SN_SIZEOF_TOKEN). We can delete StrongNameFreeBuffer and get callers to provide 8 byte buffer instead.

**Review by jkotas — APPROVED** [review]
Thank you!

**Pull request merged**